contents/javascript/Javascript-framework-Angular.md

```diff
@@ -0,0 +1,196 @@
+<frontmatter>
+  title: Introduction to Angular
+  footer: footer.md
+  head: head.md
+  siteNav: mainNav.md
+  pageNav: 3
+</frontmatter>
+
+{{ navbar | safe }}
+
+<div class="website-content">
+
+# Introduction to Angular
+
+**Author: [Ronak Lakhotia](https://github.com/RonakLakhotia)**                                                              
+Reviewers: [Tan Heng Yeow](https://github.com/tanhengyeow), [Jacob Li PengCheng](https://github.com/jacoblipech)
+
+## What is Angular?
+
+Angular, according to the [official website](https://angular.io/docs), is described as follows:
+
+>Angular is a platform that makes it easy to build applications with the web. Angular combines declarative templates, dependency injection, end to end tooling, and integrated best practices to solve development challenges. Angular empowers developers to build applications that live on the web, mobile, or the desktop.
+
+Let us understand some key terms starting with what <i>declarative templates</i> mean.
+Consider the example of creating a simple web application as shown below.
+
+![Example](First_example.png)
+
+The application displays the name and id of a person. The <i>view</i> component (i.e., the part users see) is defined in Angular using <i>templates</i>. An Angular template contains HTML code and <tooltip content="Angular allows developers to create their own HTML like tags to perform additional tasks">custom tags</tooltip>.
+Here is the template for the above example.
+  
+<iframe src="https://stackblitz.com/edit/angular-3qy4j2?embed=1&file=src/app/heroes/heroes.component.html" style="width:85%; height:380px; border:0; border-radius: 4px; overflow:hidden;" sandbox="allow-modals allow-forms allow-popups allow-scripts allow-same-origin"></iframe> 
+
+<box type="info">
+    The sandbox above is editable. You can play around with the application and see how the UI is automatically updated. 
+</box>
+
+The templates used in Angular are <i>declarative</i> because they describe <i>what</i> the view should look like rather than giving instructions on <i>how</i> to construct the view. In the above example, we declare what we want displayed (i.e., name and id) and Angular renders the view. What makes these templates unique is their dynamic behavior. You will notice that as you change the name in the text box, the name rendered on the app is also updated. This is achieved in Angular with the help of <tooltip content="The automatic synchronization between the View and Model">two-way data binding</tooltip>.
```

Comment:
`What makes these templates unique is their dynamic behavior.` This is not _unique_.

Reply:
Are you sure this is two-way data binding? If it is, is it necessary to drag it in here?
Also, I think the explanation so far could have been done without mentioning _view_ and _model_. Try to minimize jargon.

Reply:
> Are you sure this is two-way data binding? If it is, is it necessary to drag it in here?

It is because of the data binding achieved with directives. I thought of mentioning two-way data binding with a tooltip and explaining two-way data binding as an advantage later in the article

Reply:
Best not to disrupt the flow by introducing a key concepts through a tooltip.

Reply:
Yup, I removed the jargon.